packages/react/src/drafts/IssueLabel/getColorFromHex.ts

```diff
@@ -0,0 +1,137 @@
+import {getContrast} from 'color2k'
+import {Hsluv} from 'hsluv'
+
+/**
+ * transforms a hex color provided by the user into a color object with background and text colors
+ * @param colorHex — the hex color provided by the user
+ * @param colorScheme — the color scheme the user has currently set
+ * @param isSelected
+ * @param bgColor — the background color from the selected theme, needed to calc the contrast for the colors
+ * @returns
+ */
+export const getColorsFromHex = (
+  colorHex: `#{string}`,
+  colorScheme: string = 'light',
+  bgColor: string,
+): React.CSSProperties | undefined => {
+  // start values for light mode
+  let bgLightness = 96
+  let lightnessIncrement: 1 | -1 = -1
+  let ratio = 4.5
+  // start values for dark mode
+  if (colorScheme.startsWith('dark')) {
+    bgLightness = 16
+    lightnessIncrement = 1
+    ratio = 5.5
+  }
+
+  // get hue and saturation value from hex color
+  // eslint-disable-next-line prefer-const
+  let {h, s} = hexToHsluv(colorHex)
+
+  // avoid intense bright colors
+  if (h >= 58 && h <= 198 && s > 70) {
+    s = 70
+  }
+  /**
+   * creating a background color from the provided hex color
+   */
+  const {colorHex: backgroundColor, lightness: currentBgLightness} = getColorWithContrast(
+    hsluvToHex({h, s, l: bgLightness}),
+    bgColor,
+    1.2,
+    lightnessIncrement as 1 | -1,
+  )
+
+  // avoid intense bright colors
+  if (h >= 58 && h <= 316 && s > 80) {
+    s = 80
+  }
+  /**
+   * creating a text color from with a contrast ratio of at least 4.5 to the generated background color
+   */
+  const {colorHex: textColor} = getColorWithContrast(
+    hsluvToHex({h, s, l: 50}),
+    backgroundColor,
+    ratio,
+    lightnessIncrement as 1 | -1,
+  )
+
+  return {
+    '--label-bgColor-rest': backgroundColor,
+    '--label-bgColor-hover': hsluvToHex({h, s, l: currentBgLightness + 4 * lightnessIncrement}),
+    '--label-bgColor-active': hsluvToHex({h, s, l: currentBgLightness + 8 * lightnessIncrement}),
+    '--label-fgColor': textColor,
+    '--label-fgColor-hover': textColor,
+    '--label-fgColor-active': textColor,
+  } as React.CSSProperties
+  // backgroundColorHover: hsluvToHex({h, s, l: currentBgLightness + 4 * lightnessIncrement}),
+  //   backgroundColorPressed: hsluvToHex({h, s, l: currentBgLightness + 8 * lightnessIncrement}),
+}
+/**
+ * Changes the lightness of a hex color until the contrast ratio is reached and returns the new hex color
+ * @param colorHex — initial hex color
+ * @param bgHex — color to check the contrast against
+ * @param contrastRatio — the contrast ratio to reach
+ * @param increment — the direction to change the lightness depending on the color scheme
+ * @returns the new hex color
+ */
+const getColorWithContrast = (
+  colorHex: string,
+  bgHex: string,
+  contrastRatio: number,
+  increment: 1 | -1,
+): {colorHex: string; lightness: number} => {
```

Comment:
Some ideas to make these types more concrete if it's helpful:

```suggestion
const getColorWithContrast = (
  colorHex: Hex,
  bgHex: Hex,
  contrastRatio: number,
  increment: 1 | -1,
): {colorHex: Hex; lightness: number} => {
```

(Uses the `Hex` type from above)